### PR TITLE
Fix up Promises/A+ compliance. Fixes #129

### LINKED ISSUE #129: not enough async documentation/tests - BUG
i'm trying to use promises with `graph.add`

```javascript
var fs = require("q-io/fs");
 var graph = new shepherd.Graph();
 
  graph.add('config', function(){
    return fs.read("config.json").then(JSON.parse);
  });
  
  var builder = graph.newBuilder()
  .builds('config');
  
  builder
  .run({str: "Hello"})
  .then(function(data) {
    console.log(data);
  });
```

output
```
{ config: { state: 'pending' } }
```

i'm not sure why my promises aren't working with `builds()`

### PR DESCRIPTION
**Pull request by nicks**
Hello @kylehg, 

Please review the following commits I made in branch 'nick-promises'.

42a21be0128818c1ae27b707777b5c8f30bc6bf9 (2015-03-18 14:26:28 -0400)
Fix up Promises/A+ compliance. Fixes #129

R=@kylehg

### CODE REVIEW

File: test/return.js [hunk 1]
```diff
@@ -59,3 +59,21 @@ builder.add(function testResponseViaPromise(test) {
       test.equal(result.returns, response, 'Response should be returned through promise')
     })
 })
+
+// Make sure any A+-compliant promises are handled correctly.
+builder.add(function testAplusCompliant(test) {
+  this.graph.add('returns', function () {
+    return {
+      then: function (callback) {
```
Comment (boxxxie) [inline test/return.js]: why not use a real A+ promise?
bluebird/es6 
```javascript
return new Promise(function(resolve, reject){
resolve("!A+ wok");
});
```

File: lib/Builder.js [hunk 2]
```diff
@@ -2251,9 +2251,7 @@ Builder.prototype._genResolver = function (nodeName, resolveNodeFn) {
       }
     }
 
-    // TODO(kyle): Change this once kew has been updated
-    // if (Q.isPromise(result)) {
```
Comment (kylehg) [inline lib/Builder.js]: Lol thanks

**Comment by kylehg** [comment]
LG

File: test/return.js (hunk 1 above)
Comment (nicks) [inline test/return.js]: glad you asked! most promise libraries add utility methods. I wanted to make sure we didn't call any of the new methods accidentally. this is a minimally viable promise, per the spec

**Pull request merged**